src/resources/js/move-tickets-dialog.js

```diff
@@ -0,0 +1,496 @@
+/**
+ * Handles the move tickets/move ticket type dialogs.
+ *
+ * @var object tribe_move_tickets
+ * @var object tribe_move_tickets_data
+ */
+
+var tribe_move_tickets = ( 'object' === typeof tribe_move_tickets ) ? tribe_move_tickets : {};
+
+( function( $, obj ) {
+	var $main,
+		$processing,
+		$errors,
+		$current_stage,
+		current_stage_name,
+		$stages,
+		$back,
+		$next,
+		last_direction,
+		skip_choose_event,
+		target_post_id,
+		target_ticket_type_id;
+
+
+	function init() {
```

Comment:
Our _PHP_ coding standards state that methods should be private by default and I suppose there is merit to our JS following suit, though that line hasn't been explicitly drawn in the sand to this point.

Perhaps I have a jaded view of plugins/services that provide JS that isn't easily modifiable/easy to override. In my previous positions privately scoped third-party JS was the bane of my existence when tasked with mods/complex behaviors for products I was working on. There's probably a pleasant middle ground (and your third bullet is most likely it).

I guess my thoughts on your questions are:

* Perhaps we don't have to. We can aim to expose stuff that _should_ be exposed and keep private the stuff that needs to be private.
* If our goal is to avoid customizations of the wizard, then I suppose not. As it stands, if I wanted to inject an additional stage into the wizard and utilize some of the same button behaviors that are defined, that would prove troublesome (unless I'm missing something). That's probably ok for now and we can just wait for a feature request which may never come!
* Events are great.